src/transports/smooth/pipelines.ts

```diff
@@ -0,0 +1,443 @@
+/**
+ * Copyright 2015 CANAL+ Group
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  Observable,
+  of as observableOf,
+} from "rxjs";
+import { map } from "rxjs/operators";
+import features from "../../features";
+import log from "../../log";
+import Manifest, {
+  Adaptation,
+  Representation,
+} from "../../manifest";
+import createSmoothManifestParser from "../../parsers/manifest/smooth";
+import assert from "../../utils/assert";
+import request from "../../utils/request";
+import { stringFromUTF8 } from "../../utils/strings";
+import warnOnce from "../../utils/warn_once";
+import {
+  ILoaderObservable,
+  ImageParserObservable,
+  IManifestLoaderArguments,
+  IManifestParserArguments,
+  IManifestParserObservable,
+  INextSegmentsInfos,
+  ISegmentLoaderArguments,
+  ISegmentParserArguments,
+  ISegmentTimingInfos,
+  ITransportOptions,
+  ITransportPipelines,
+  SegmentParserObservable,
+  TextTrackParserObservable,
+} from "../types";
+import generateManifestLoader from "../utils/manifest_loader";
+import extractTimingsInfos from "./extract_timings_infos";
+import mp4Utils from "./mp4_utils";
+import generateSegmentLoader from "./segment_loader";
+import {
+  extractISML,
+  extractToken,
+  replaceToken,
+  resolveManifest,
+} from "./utils";
+
+const {
+  patchSegment,
+  getMdat,
+} = mp4Utils;
+
+const WSX_REG = /\.wsx?(\?token=\S+)?/;
+
+/**
+ * @param {Object} adaptation
+ * @param {Object} dlSegment
+ * @param {Object} nextSegments
+ */
+function addNextSegments(
+  adaptation : Adaptation,
+  nextSegments : INextSegmentsInfos[],
+  dlSegment? : ISegmentTimingInfos
+) : void {
+  log.debug("Smooth Parser: update segments informations.");
+  const representations = adaptation.representations;
+  for (let i = 0; i < representations.length; i++) {
+    const representation = representations[i];
+    representation.index._addSegments(nextSegments, dlSegment);
+  }
+}
+
+export default function(
+  options : ITransportOptions = {}
+) : ITransportPipelines {
+  const smoothManifestParser = createSmoothManifestParser(options);
+  const segmentLoader = generateSegmentLoader(options.segmentLoader);
+
+  const manifestLoaderOptions = {
+    customManifestLoader: options.manifestLoader,
+    ignoreProgressEvents: true as true,
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove this unnecessary cast. ([typescript:S4325](https://sonarcloud.io/organizations/rx-player/rules#rule_key=typescript%3AS4325))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=rx-player&pullRequest=380&issues=AWflYNhn_MPaBDn0rlBf&open=AWflYNhn_MPaBDn0rlBf)

[ISSUE_KEY]: AWflYNhn_MPaBDn0rlBf